modules/Forum/classes/Forum.php

```diff
@@ -808,4 +797,47 @@ public static function getAccessibleLabels(array $labels, array $user_groups): a
             return $prev;
         }, []);
     }
+
+    /**
+     * Get the latest post in a "View own topic" forum
+     * This could be a topic created by the user, or a sticky topic
+     *
+     * @param int $forumId
+     * @param int $userId
+     * @return array|null Time of latest post, post creator ID, topic ID
+     */
+    private function getLatestPostInOwnTopicForum(int $forumId, int $userId): ?array {
+        $latest_post = $this->_db->query(
+            <<<SQL
+                SELECT
+                    p.topic_id,
+                    p.created,
+                    p.post_date,
+                    p.post_creator
+                FROM nl2_topics t
+                    LEFT JOIN nl2_posts p
+                        ON p.id = (
+                            SELECT id
+                            FROM nl2_posts sp
+                            WHERE sp.topic_id = t.id
+                            AND sp.deleted = 0
+                            ORDER BY sp.created DESC LIMIT 1
+                        )
+                WHERE t.forum_id = ?
+                  AND (t.topic_creator = ? OR t.sticky = 1)
+                ORDER BY p.created, p.post_date DESC LIMIT 1
+            SQL,
+            [$forumId, $userId]
+        );
+
+        if ($latest_post->count() && $latest_post = $latest_post->first()) {
+            return [
+                $latest_post->created ?? strtotime($latest_post->post_date),
```

Comment:
Out of scope for this PR but this definitely feels like code smell

Reply:
Can I mark this conversation as resolved to be able to merge this PR?